src/Compute/Compute/Extension/VmssDiskEncryption/SetAzureVmssDiskEncryptionExtension.cs

```diff
@@ -115,11 +115,31 @@ public class SetAzureVmssDiskEncryptionExtensionCommand : VirtualMachineScaleSet
         [Parameter(
             Mandatory = false,
             ValueFromPipelineByPropertyName = true,
-            HelpMessage = "The extension name. If this parameter is not specified, default values used are AzureDiskEncryption for windows VMs and AzureDiskEncryptionForLinux for Linux VMs")]
+            HelpMessage = "The extension publisher name. If this parameter is not specified, default value is Microsoft.Azure.Security for Windows VMs and Linux VMs")]
+        [ResourceNameCompleter("Microsoft.Compute/virtualMachineScaleSets/extensions", "ResourceGroupName", "VMScaleSetName")]
```

Comment:
`ResourceNameCompleter` can only be used when the thing you are trying to complete _is a resource_. Otherwise please use `ArgumentCompleter`